server/create.go

```diff
@@ -0,0 +1,645 @@
+package server
+
+import (
+	"bytes"
+	"cmp"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/ollama/ollama/api"
+	"github.com/ollama/ollama/convert"
+	"github.com/ollama/ollama/envconfig"
+	"github.com/ollama/ollama/format"
+	"github.com/ollama/ollama/llama"
+	"github.com/ollama/ollama/llm"
+	"github.com/ollama/ollama/template"
+	"github.com/ollama/ollama/types/errtypes"
+	"github.com/ollama/ollama/types/model"
+)
+
+func (s *Server) CreateHandler(c *gin.Context) {
```

Comment:
Optional build would be awesome to make this (and future handlers) a `http.HandlerFunc` so we can slowly move to `http.ServeMux` in `routes.go`

Reply:
Ah might be hard given we're using Gin's streaming stuff later on in the func. Up to you